src/java.base/share/classes/java/nio/MappedByteBuffer.java

```diff
@@ -284,7 +288,15 @@ public final MappedByteBuffer force(int index, int length) {
         if ((address != 0) && (limit != 0)) {
             // check inputs
             Objects.checkFromIndexSize(index, length, limit);
-            SCOPED_MEMORY_ACCESS.force(scope(), fd, address, isSync, index, length);
+            try {
+                SCOPED_MEMORY_ACCESS.force(scope(), fd, address, isSync, index, length);
+            } catch (Exception unspecifiedException) {
```

Comment:
The force method should be changed to declare that it throws IOException and that will avoid the need to catch "unexpectedException" here.

Also I think we to need to decide if the force methods should be specified to throw UncheckedIOException rather than an "unspecified exception". The API changes will need to be coordinated with Project Panama and the MappedMemorySegments.force spec.

Reply:
In commit [df506b9](https://github.com/openjdk/jdk/pull/2636/commits/df506b9b09d6545982ed946d13737d5829863932), `MappedMemoryUtils.force()` is modified to rethrow an `IOException` thrown by `MappedMemoryUtils.force0()` as an `UncheckedIOException` with the `IOException` as its cause. The specifications of `force()` in `MappedByteBuffer` and `MappedMemorySegments` are modified to specify that such an `UncheckedIOException` will be thrown if the underlying operation fails.